### PR TITLE
Add "Look up DOIs" to Quality menu

### PR DESCRIPTION
**Pull request by koppor**
This adds "Lookup DOIs" to the quality menu. Additionally: Rename lookup to look up (see http://notaverb.com/lookup)

With an updating status bar:

grabbed_20161231-184622

- [x] Change in CHANGELOG.md described
- [ ] Tests created for changes
- [x] Screenshots added (for bigger UI changes)
- [x] Manually tested changed features in running JabRef
- [ ] Check documentation status (Issue created for outdated help page at [help.jabref.org](https://github.com/JabRef/help.jabref.org/issues)?)
- [x] If you changed the localization: Did you run `gradle localizationUpdate`?

### CODE REVIEW

File: src/main/java/net/sf/jabref/gui/worker/LookupDOIsWorker.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+            String totalCount = Integer.toString(bibEntries.size());
+            NamedCompound namedCompound = new NamedCompound(Localization.lang("Look up DOIs"));
+            int count = 0;
+            int foundCount = 0;
+            for (BibEntry bibEntry : bibEntries) {
+                count++;
+                frame.output(Localization.lang("Looking up DOIs... - entry %0 out of %1 - found %2", Integer.toString(count), totalCount, Integer.toString(foundCount)));
+                if (!bibEntry.hasField(FieldName.DOI)) {
+                    Optional<DOI> doi = DOI.fromBibEntry(bibEntry);
+                    if (doi.isPresent()) {
```
Comment (Siedlerchr) [inline src/main/java/net/sf/jabref/gui/worker/LookupDOIsWorker.java]: You could chain the optionals with flatmap, avoid the ifPresent cascading...

http://www.nurkiewicz.com/2013/08/optional-in-java-8-cheat-sheet.html

File: src/main/java/net/sf/jabref/gui/worker/LookupDOIsWorker.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+import org.apache.commons.logging.LogFactory;
+
+public class LookupDOIsWorker extends AbstractWorker {
+
+    private final JabRefFrame frame;
+    private String message;
+
+    private static final Log LOGGER = LogFactory.getLog(LookupDOIsWorker.class);
+
+    public LookupDOIsWorker(JabRefFrame frame) {
```
Comment (tobiasdiez) [inline src/main/java/net/sf/jabref/gui/worker/LookupDOIsWorker.java]: For maximal reusability, could you please make this class independent on DOI. There are many other identifier for which such an automatic look-up makes sense.

Comment (koppor) [inline src/main/java/net/sf/jabref/gui/worker/LookupDOIsWorker.java]: - [ ] Language: Either `ID` or let the caller set the DOI name
- [ ] lines 43 to 46 somehow extract. Interface: CanCompleteEntry

File: src/main/java/net/sf/jabref/gui/worker/LookupDOIsWorker.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    @Override
+    public void run() {
+        BasePanel basePanel = Objects.requireNonNull(frame.getCurrentBasePanel());
+        List<BibEntry> bibEntries = basePanel.getSelectedEntries();
+        if (!bibEntries.isEmpty()) {
+            String totalCount = Integer.toString(bibEntries.size());
+            NamedCompound namedCompound = new NamedCompound(Localization.lang("Look up DOIs"));
+            int count = 0;
+            int foundCount = 0;
+            for (BibEntry bibEntry : bibEntries) {
```
Comment (koppor) [inline src/main/java/net/sf/jabref/gui/worker/LookupDOIsWorker.java]: I have no idea how to create a progress bar there.

I found http://stackoverflow.com/a/26814087/873282, but that code looks much more complicated than the current code. Should I change it?

File: src/main/java/net/sf/jabref/gui/worker/LookupDOIsWorker.java (hunk 1 above)
Comment (koppor) [inline src/main/java/net/sf/jabref/gui/worker/LookupDOIsWorker.java]: No clue how I can do that with progress...

File: src/main/java/net/sf/jabref/gui/worker/LookupDOIsWorker.java (hunk 3 above)
Comment (Siedlerchr) [inline src/main/java/net/sf/jabref/gui/worker/LookupDOIsWorker.java]: Na then go for it now.,

File: src/main/java/net/sf/jabref/gui/worker/LookupDOIsWorker.java (hunk 1 above)
Comment (Siedlerchr) [inline src/main/java/net/sf/jabref/gui/worker/LookupDOIsWorker.java]: Okay, was just an idea. If it does not make it easier forget about it

File: src/main/java/net/sf/jabref/gui/worker/LookupDOIsWorker.java (hunk 2 above)
Comment (tobiasdiez) [inline src/main/java/net/sf/jabref/gui/worker/LookupDOIsWorker.java]: @koppor can you please rebase and I will implement then the rest. Deal?

**Comment by tobiasdiez** [comment]
@koppor thanks for rebased. I generalized it now to also work for other identifiers so this PR is now ready for review for me.

File: src/main/java/org/jabref/model/entry/identifier/Identifier.java [hunk 4]
```diff
@@ -0,0 +1,8 @@
+package org.jabref.model.entry.identifier;
+
+public interface Identifier {
```
Comment (koppor) [inline src/main/java/org/jabref/model/entry/identifier/Identifier.java]: Should we call this class HasIdentifier?

**Comment by koppor** [comment]
LGTM 😇 

I get multiple errors when using it. I don't get the information which entry is affected. But this is not an issue of this PR. Just wanted to note it down somewhere.

`2017-03-17 18:14:36,528 Spin-0 ERROR Recursive call to appender GuiLogger`

```
Caused by: org.jabref.logic.importer.ParseException: CrossRef API JSON format has changed
        at org.jabref.logic.importer.fetcher.CrossRef.jsonItemToBibEntry(CrossRef.java:131) ~[main/:?]
        at org.jabref.logic.importer.fetcher.CrossRef.lambda$getParser$3(CrossRef.java:88) ~[main/:?]
        at org.jabref.logic.importer.IdParserFetcher.findIdentifier(IdParserFetcher.java:75) ~[main/:?]
        ... 8 more
Caused by: org.json.JSONException: JSONObject["given"] not found.
        at org.json.JSONObject.get(JSONObject.java:471) ~[json-20160212.jar:?]
        at org.json.JSONObject.getString(JSONObject.java:717) ~[json-20160212.jar:?]
        at org.jabref.logic.importer.fetcher.CrossRef.toAuthors(CrossRef.java:144) ~[main/:?]
        at org.jabref.logic.importer.fetcher.CrossRef.jsonItemToBibEntry(CrossRef.java:117) ~[main/:?]
        at org.jabref.logic.importer.fetcher.CrossRef.lambda$getParser$3(CrossRef.java:88) ~[main/:?]
        at org.jabref.logic.importer.IdParserFetcher.findIdentifier(IdParserFetcher.java:75) ~[main/:?]
```

`Caused by: java.io.IOException: Server returned HTTP response code: 500 for URL: http://api.crossref.org/works?query.title=A+%28sub%29graph+isomorphism+algorithm+for+matching+large+graphs&query.author=L.P.+Cordella+and+P.+Foggia+and+C.+Sansone+and+M.+Vento&filter=from-pub-date%3A2004&rows=20&offset=0`

**Pull request merged**